Writerside/topics/introduction.md

```diff
@@ -1,97 +1,82 @@
+for such a task.
+
+These research questions are framed within the context of the current state of public transit networks. The complexity
+of these networks, which involve multiple routes, stops, footpaths, and transfer points, necessitates the development of
+routing algorithms that account for factors such as transfer times, service frequencies, and real-time constraints. The
+research seeks to address these challenges while also considering scalability, as expanding transit systems further
+complicates the problem space.
+
+## Approach and Expected Outcomes
+
+This thesis adopts a comprehensive approach to address the identified challenges, beginning with the utilization of GTFS
+data as the foundation for the transit routing system. The standardized format of GTFS [3], which includes
+static information such as stop times, routes, and trips, provides a consistent and reliable dataset for the
+implementation of routing algorithms. A memory-efficient library for reading and processing GTFS schedules will be
+developed to enable scalable management of large datasets, ensuring the system can handle large-scale transit networks
+effectively.
+
+One of the core aspects of this thesis is the implementation of the RAPTOR algorithm [4], which is
+designed to efficiently compute transit routes by considering time dependencies such as schedules and transfer windows.
+The RAPTOR algorithm employs a round-based approach, where transit routes are organized into discrete rounds, each
+representing the number of trips needed. This structure simplifies the traversal of the network by enabling the
+algorithm to explore transit options in rounds and return pareto-optimal solutions, minimizing both arrival time and the
+number of transfers.
+
+The system will be complemented with an integration layer that manages both the routing algorithm and the schedule,
+while handling all necessary conversions between the two. This design maintains the independence of the schedule and the
+routing algorithm, enabling a modular architecture. The modularity allows for system components to be modified or
+replaced without affecting external dependencies, ensuring the system's adaptability and long-term maintainability.
+
+To facilitate interaction with the system, a REST API will be developed, allowing external applications to access the
+routing and scheduling services. This API will provide the foundation for seamless integration between the routing
+algorithm and external systems. As a proof of concept, a simple web application will be created to visualize search
+results and demonstrate the capabilities of the backend service. While the web application will serve as a proof of
+concept, the primary focus of this thesis remains on developing the backend service.
+
+This approach ensures the development of a scalable, high-performance public transit routing package based on the RAPTOR
```

Comment:
Maybe remove this paragraph? It is kind of a repetition of the above?

Reply:
Agreed, 0a3dc1b